CoreImportExportPlugins/test/unit/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/parser/CSVImportFileParserNGTest.java

```diff
@@ -0,0 +1,169 @@
+    private static PluginParameters pluginParametersMock;
+    private static CSVParser CSVParserMock;
+    private static Iterator<CSVRecord> iteratorMock;
+    private static CSVRecord CSVRecordMock;
+
+    public CSVImportFileParserNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        inputSourceMock = Mockito.mock(InputSource.class);
+        pluginParametersMock = Mockito.mock(PluginParameters.class);
+        CSVParserMock = Mockito.mock(CSVParser.class);
+        iteratorMock = Mockito.mock(Iterator.class);
+        CSVRecordMock = Mockito.mock(CSVRecord.class);
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of parse method, of class CSVImportFileParser.
+     *
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testParse() throws Exception {
+        System.out.println("testParse");
+
+        final CSVImportFileParser instance = spy(new CSVImportFileParser());
+
+        // When the CSV file is empty.
+        doReturn(CSVParserMock).when(instance).getCSVParser(inputSourceMock);
+        doReturn(iteratorMock).when(CSVParserMock).iterator();
+
+        doCallRealMethod().when(instance).parse(inputSourceMock, pluginParametersMock);
+
+        final List expResult1 = new ArrayList<>();
+        final List result1 = instance.parse(inputSourceMock, pluginParametersMock);
+
+        assertEquals(result1, expResult1);
+
+        // When there are CSV records to be parsed in the file.
+        doReturn(Boolean.TRUE, Boolean.TRUE, Boolean.FALSE).when(iteratorMock).hasNext();
+        doReturn(CSVRecordMock, CSVRecordMock).when(iteratorMock).next();
+        doReturn(1).when(CSVRecordMock).size();
+        doReturn("test").when(CSVRecordMock).get(0);
+
+        final String[] line = new String[1];
+        line[0] = "test";
+
+        final ArrayList<String[]> list = new ArrayList<>();
+        list.add(line);
+        list.add(line);
+
+        final List expResult2 = list;
+        final List result2 = instance.parse(inputSourceMock, pluginParametersMock);
+
+        assertEquals(result2, expResult2);
+    }
+
+    /**
+     * Test of preview method, of class CSVImportFileParser.
+     *
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testPreview() throws Exception {
+        System.out.println("testPreview");
+
+        final CSVImportFileParser instance = spy(new CSVImportFileParser());
+
+        // When the CSV file is empty.
+        doReturn(CSVParserMock).when(instance).getCSVParser(inputSourceMock);
+        doReturn(iteratorMock).when(CSVParserMock).iterator();
+
+        // The limit value is irrelevant in this case.
+        final Random rand = new Random();
+        final int limit = rand.nextInt(50) + 1;
```

Comment:
if the limit is irrelevant, could you just pick an arbitrary number yourself rather than randomly generated one? Else if necessary, use `SecureRandom`